### PR TITLE
add interaction field to message struct

### PR DESCRIPTION
**Pull request by TopiSenpai**
This pr adds the interaction field to the message struct 
https://discord.com/developers/docs/resources/channel#message-object-message-structure

there is another pr for this which seems to be dead and is wrongly implemented https://github.com/bwmarrin/discordgo/pull/929

### CODE REVIEW

File: message.go [hunk 1]
```diff
@@ -135,6 +135,11 @@ type Message struct {
 	// If the field exists but is null, the referenced message was deleted.
 	ReferencedMessage *Message `json:"referenced_message"`
 
+	// MessageInteraction is sent on the message object when the message is a response to an Interaction without an existing message.
+	// This means responses to Message Components do not include this property,
+	// instead including a MessageReference as components always exist on preexisting messages.
```
Comment (FedorLap2006) [inline message.go:140]:
```suggestion
	// Is sent when the message is a response to an Interaction, without an existing message.
	// This means responses to message component interactions do not include this property,
	// instead including a MessageReference, as components exist on preexisting messages.
```

File: message.go [hunk 2]
```diff
@@ -523,3 +528,12 @@ func (m *Message) ContentWithMoreMentionsReplaced(s *Session) (content string, e
 	})
 	return
 }
+
+// MessageInteraction contains information about the application command interaction which generated this message
```
Comment (FedorLap2006) [inline message.go:532]:
```suggestion
// MessageInteraction contains information about the application command interaction which generated a message
```

File: message.go [hunk 3]
```diff
@@ -523,3 +528,12 @@ func (m *Message) ContentWithMoreMentionsReplaced(s *Session) (content string, e
...
+type MessageInteraction struct {
+	ID     string          `json:"id"`
+	Type   InteractionType `json:"type"`
+	Name   string          `json:"name"`
+	User   User            `json:"user"`
```
Comment (FedorLap2006) [inline message.go:537]:
```suggestion
	User   *User            `json:"user"`
```

**Review by FedorLap2006 — CHANGES_REQUESTED** [review]

File: message.go (hunk 3 above)
Comment (TopiSenpai) [reply message.go:537]: as far as I can tell the user field is always present
https://discord.com/developers/docs/interactions/receiving-and-responding#message-interaction-object-message-interaction-structure

**Pull request merged**